src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -73,53 +74,44 @@ internal void Clear()
+		private static IDisposable InnerCreateLayer(UIElement owner, LayoutState state)
 		{
+			var parent = owner.Visual;
+			var compositor = parent.Compositor;
+			var area = owner.LayoutRound(state.Area);
+			var background = state.Background;
+			var borderThickness = state.BorderThickness;
+			var borderBrush = state.BorderBrush;
+			var cornerRadius = state.CornerRadius;
+
 			var disposables = new CompositeDisposable();
-			var subVisuals = new List<Visual>();
+			var sublayers = new List<Visual>();
 
-			var adjustedLineWidth = borderThickness.Top;
-			var adjustedLineWidthOffset = adjustedLineWidth / 2;
+			var adjustedStrokeThickness = borderThickness.Top;
+			var adjustedStrokeThicknessOffset = adjustedStrokeThickness / 2;
 
 			var adjustedArea = area;
-			adjustedArea.Inflate(-adjustedLineWidthOffset, -adjustedLineWidthOffset);
-
-			var compositor = parent.Compositor;
+			adjustedArea = adjustedArea.DeflateBy(new Thickness(adjustedStrokeThicknessOffset));
 
-			var shapeVisual = compositor.CreateShapeVisual();
-			parent.Children.InsertAtBottom(shapeVisual);
-
-			var spriteShape = compositor.CreateSpriteShape();
-
-			SkiaGeometrySource2D BuildGeometry()
+			if (cornerRadius != CornerRadius.None)
 			{
-				var maxRadius = Math.Max(0, Math.Min((float)area.Width / 2 - adjustedLineWidthOffset, (float)area.Height / 2 - adjustedLineWidthOffset));
+				var maxRadius = Math.Max(0, Math.Min((float)area.Width / 2 - adjustedStrokeThicknessOffset, (float)area.Height / 2 - adjustedStrokeThicknessOffset));
 				cornerRadius = new CornerRadius(
 					Math.Min(cornerRadius.TopLeft, maxRadius),
 					Math.Min(cornerRadius.TopRight, maxRadius),
 					Math.Min(cornerRadius.BottomRight, maxRadius),
 					Math.Min(cornerRadius.BottomLeft, maxRadius));
 
-				var geometry = new SkiaGeometrySource2D();
+				var spriteShape = compositor.CreateSpriteShape();
+				spriteShape.StrokeThickness = (float)adjustedStrokeThickness;
 
 				Brush.AssignAndObserveBrush(borderBrush, color => spriteShape.StrokeBrush = compositor.CreateColorBrush(color))
 					.DisposeWith(disposables);
+				var path = GetRoundedPath(cornerRadius, adjustedArea);
 
-				geometry.Geometry.MoveTo((float)adjustedArea.GetMidX(), (float)adjustedArea.Y);
-				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Top, (float)adjustedArea.Right, (float)adjustedArea.GetMidY(), (float)cornerRadius.TopRight);
-				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Bottom, (float)adjustedArea.GetMidX(), (float)adjustedArea.Bottom, (float)cornerRadius.BottomRight);
-				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Bottom, (float)adjustedArea.Left, (float)adjustedArea.GetMidY(), (float)cornerRadius.BottomLeft);
-				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Top, (float)adjustedArea.GetMidX(), (float)adjustedArea.Top, (float)cornerRadius.TopLeft);
+				var outerPath = GetRoundedPath(cornerRadius, area);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'outerPath'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5924508)